src/tools/miri/tests/pass/float.rs

```diff
@@ -1144,6 +1144,46 @@ fn test_algebraic() {
         fadd_algebraic, fdiv_algebraic, fmul_algebraic, frem_algebraic, fsub_algebraic,
     };
 
+    #[inline(never)]
+    pub fn test_operations_f16(a: f16, b: f16) {
+        // make sure they all map to the correct operation
+        assert_eq!(a.algebraic_add(b), a + b);
+        assert_eq!(a.algebraic_sub(b), a - b);
+        assert_eq!(a.algebraic_mul(b), a * b);
+        assert_eq!(a.algebraic_div(b), a / b);
+        assert_eq!(a.algebraic_rem(b), a % b);
+    }
+
+    #[inline(never)]
+    pub fn test_operations_f32(a: f32, b: f32) {
+        // make sure they all map to the correct operation
+        assert_eq!(a.algebraic_add(b), a + b);
+        assert_eq!(a.algebraic_sub(b), a - b);
+        assert_eq!(a.algebraic_mul(b), a * b);
+        assert_eq!(a.algebraic_div(b), a / b);
+        assert_eq!(a.algebraic_rem(b), a % b);
+    }
+
+    #[inline(never)]
+    pub fn test_operations_f64(a: f64, b: f64) {
+        // make sure they all map to the correct operation
+        assert_eq!(a.algebraic_add(b), a + b);
+        assert_eq!(a.algebraic_sub(b), a - b);
+        assert_eq!(a.algebraic_mul(b), a * b);
+        assert_eq!(a.algebraic_div(b), a / b);
+        assert_eq!(a.algebraic_rem(b), a % b);
+    }
+
+    #[inline(never)]
+    pub fn test_operations_f128(a: f128, b: f128) {
+        // make sure they all map to the correct operation
+        assert_eq!(a.algebraic_add(b), a + b);
+        assert_eq!(a.algebraic_sub(b), a - b);
+        assert_eq!(a.algebraic_mul(b), a * b);
+        assert_eq!(a.algebraic_div(b), a / b);
+        assert_eq!(a.algebraic_rem(b), a % b);
+    }
+
```

Comment:
I figured it was a good idea to add these, but happy to remove or update the existing ones to use the stable API instead.

Reply:
We already have `test_algebraic` so this feels a bit redundant... also, the assertion will fail soon when https://github.com/rust-lang/miri/pull/4156 lands.

Reply:
Reverted.

Reply:
@RalfJung: Should we at least update these tests to use the public API? It feels weird to only test the internal API but not the public one.

Reply:
Ultimately the intrinsics are what Miri needs to implement. The doctests in this repo (most of them, anyway) are run in Miri as well so we should notice if doctests start using a different intrinsic we do not support yet.